Updated submission logic to treat only CRIT1 & FATAL as critical errors (Crit 2 submissions #6629 & #6539)

spetros-do opened the pull request:
### Changes Made: 
- **Submission Process Service**: Modified to directly check for CRIT1 and FATAL severity codes instead of relying on the UI-provided `hasCritErrors` flag
- **Document Service**: Updated to only generate evaluation reports for CRIT1 and FATAL severity codes
- **Submission Feedback Record Service**: Updated to only require resubmission for CRIT1 and FATAL severity codes
- **Submission Email Parameters DTO**: Updated helper functions to only consider CRIT1 and FATAL as critical
 
## Testing
- Verified that submissions with CRIT2 and CRIT3 severity codes can be processed successfully
- Confirmed that submissions with CRIT1 and FATAL severity codes are still blocked from processing
- Validated that feedback reports correctly indicate which records require 

## Reference to #6629 & #6539

spetros-do committed:
Updated submission logic to treat only CRIT1 & FATAL as critical errors; updated feedback handling and modified copyToOfficial to check severity codes directly.

spetros-do commented:
![db_servityCodeTable](https://github.com/user-attachments/assets/31faaefe-e62e-4bce-96a8-fee246949bc5)

yonatan-dp commented on src/dto/submission-email-params.dto.ts:
```diff
@@ -42,7 +42,7 @@ export function isCritical1Severity(record: HighestSeverityRecord): boolean {
 
 export function isResubmissionRequired(record: HighestSeverityRecord): boolean {
-  return record?.submissionQueue?.severityCode?.toUpperCase() === 'CRIT1' || record?.submissionQueue?.severityCode?.toUpperCase() === 'CRIT2';
+  return record?.submissionQueue?.severityCode?.toUpperCase() === 'CRIT1' || record?.submissionQueue?.severityCode?.toUpperCase() === 'FATAL';
```
I am not 100% sure the resubmission flag on the feedback reports is also impacted here.  Ticket #6080 specifically states that resubmission is required for CRIT1 and CRIT2 errors. @djw4erg thinks so as well. Can we confirm this with @esaber76?

Here is the relevant section from that ticket (https://github.com/US-EPA-CAMD/easey-ui/issues/6080)

`The Resubmission Required value is based on the SEVERITY_CD value for the CHECK_SESSION associated with the data. If the SEVERITY_CD is 'CRIT1' or 'CRIT2', resubmission is required.`

djw4erg replied:
Agreed.  Resubmission required is not the same as can copy to Official.

spetros-do replied:
Updated. Thx

yonatan-dp commented on src/submission/document.service.ts:
```diff
@@ -50,8 +50,9 @@ export class DocumentService {
-      if (['CRIT1', 'CRIT2', 'CRIT3', 'FATAL'].includes(rec.severityCode)) {
-        const params = new ReportParamsDTO();
+      // Only generate evaluation reports for critical severity codes (CRIT1 and FATAL)
+      if (['CRIT1', 'FATAL'].includes(rec.severityCode)) {
```
Similar to the comments above, I think we should still be adding evaluation reports based on, not just CRIT1 and FATAL.  @djw4erg can comment on this to confirm.

djw4erg replied:
The purpose of this ticket is to resolve the following issues from the background in the original ticket:

- I submitted a MP with only a critical 2 error. The MP updates were not moved to official. After the submission and once I reverted, the changes were lost and the original MP was set back to normal. This is not correct behavior.

- For Crit 2 submissions, users should not have to select the "Include Files with Critical Errors" option in the Submit screen to perform Crit 2 submissions.

Changing the documents included in the CDX submission are not part of this ticket, and would likely introduce a bug.

spetros-do replied:
Updated. Thx

yonatan-dp commented on src/submission/submission-feedback-record.service.ts:
```diff
@@ -92,7 +92,7 @@ export class SubmissionFeedbackRecordService {
     }
     const severityCode = highestSeverityRecord.severityCode.severityCode;
-    return severityCode === 'CRIT1' || severityCode === 'CRIT2';
+    return severityCode === 'CRIT1' || severityCode === 'FATAL';
```
Same comment. This original logic is based on Dwayne's comment on Ticket #6080 

`The Resubmission Required value is based on the SEVERITY_CD value for the CHECK_SESSION associated with the data. If the SEVERITY_CD is 'CRIT1' or 'CRIT2', resubmission is required.`

djw4erg replied:
Agreed.

spetros-do replied:
Updated. Thx

yonatan-dp commented:
@spetros-do, I do not see the following changes in camd-services based on the updated requirements.  Just copy pasted these from the updated requirements on the ticket...
 - Ignore the incoming `submissionQueueParam.hasCritErrors` parameter in `SubmissionService.queueSubmissionRecords(...)`. 
- Remove `hasCritErrors` attribute from the relevant DTOs (e.g. `SubmissionQueueDTO`).
- Ensure CAMD Services no longer relies on this flag being passed in by the ECMPS UI.
- Write new logic to set the `SubmissionSet.hasCritErrors` flag on the submission set object independently during queueing. It should no longer rely on the incoming `submissionQueueParam.hasCritErrors` attribute.

yonatan-dp commented:
@spetros-do, please ignore this comment if you have assigned the changes to db-scripts and ecmps-ui to another developer(s). But, I don't see those PRs linked to the ticket and I was going to review those too to make sure they line up with camd-services changes.

djw4erg reviewed (changes requested):
## Of Note

There are really two parts to a submission:

1. The submission of MP, QA or EM report files with evaluation reports and other documentation to CDX.
2. The copying of MP, QA or EM data from Workspace to Official.

This ticket was intended to allow the copying to Official for CRIT2 reports, and to not require selecting "Include Files with Critical Errors" to submit CRIT2 reports.  Anything else is outside the intent for this ticket.  This is especially true if it affects parts of the CDX submission.

spetros-do committed:
Merge remote-tracking branch 'origin/develop' into fix/#6629-critical-error-handing-in-submission-process

spetros-do committed:
Merge origin/develop into fix/#6629 and clean up files after review

spetros-do commented:
My initial refactoring was based our team meeting. However, after reviewing @yonatan-dp & @djw4erg comments; I have reverted the changes to the original. Hence, the only change made that will directly address the Crit2 Errors is the refactoring done to the submission processing as following: 
## Changes Made: 
- Modified `copyToOfficial` method in `submission-process.service.ts` to directly check for CRIT1 and FATAL severity codes.
- Updated corresponding test file to verify the new behavior.
- No UI changes required as the frontend doesn't enforce any specific behavior for CRIT2 errors.
## Purpose & Impact: 
- This change streamlines the submission process by allowing submissions with CRIT2 errors to be processed without requiring users to select the "Include Files with Critical Errors" option. 
- Only submissions with CRIT1 and FATAL severity codes will be blocked from copying to the official database.
- For Crit 2 submissions, users do not have to select the "Include Files with Critical Errors" option in the Submit screen to perform Crit 2 submissions.
- No UI changes are needed because the UI doesn't enforce any specific behavior for CRIT2 errors - it was the backend that was blocking these submissions, which are now fixed.

## Testing:
- Unit tests have been updated to verify that:
  - Submissions with CRIT1 or FATAL severity codes are blocked.
  - Submissions with CRIT2 errors are allowed to proceed.
  - Error handling remains intact.

spetros-do commented:
Replaced with PR #360
